virt_lightning/virt_lightning.py

```diff
@@ -252,14 +252,45 @@ def prepare_cloud_init_openstack_iso(self, domain):
                 "uuid": domain.dom.UUIDString(),
                 "admin_pass": domain.root_password,
             }
+            links = []
+            additional_networks = []
+            for i, addr in enumerate(domain.mac_addresses):
+                link = {
+                    "id": "interface" + str(i),
+                    "type": "phy",
+                    "ethernet_mac_address": addr,
+                }
+                links.append(link)
+            if domain.additional_ipv4 != None:
+                ips = domain.additional_ipv4.split(",")
+                for i, ipv4 in enumerate(ips):
+                    if ipv4 == "none":
+                        continue
+                    if ipv4 == "dhcp":
+                        net = {
+                            "id": "private-ipv4-" + str(i),
+                            "type": "ipv4_dhcp",
+                            "link": "interface" + str(i + 1),
+                            "network_id": "da5bb487-5193-4a65-a3df-4a0055a8c0d7",
+                        }
+                        additional_networks.append(net)
+                    else:
+                        netmask = "255.255.255.0"
```

Comment:
I suggest a slightly different approach. Pseudo code:

if '/' not in ipv4:
    ipv4 += '/24'
ip_instance = ipaddress.IPv4Interface(ipv4)

And then, you can use `ip_instance.ip` and `ip_instance.netmask.exploded`:

```
>>> my_ip = ipaddress.IPv4Interface('192.168.123.4/24')
>>> my_ip.ip
IPv4Address('192.168.123.4')
>>> my_ip.netmask.exploded
'255.255.255.0'
```